Extensions/JPLibffi/JPMethodSignature.m

```diff
@@ -155,6 +173,27 @@ + (ffi_type *)ffiTypeWithEncodingChar:(const char *)c
         return &ffi_type_pointer;
         case '#':
         return &ffi_type_pointer;
+        case '{':
+        {
+            NSString *typeStr = [NSString stringWithCString:c encoding:NSASCIIStringEncoding];
+            NSString *structName = [typeStr substringWithRange:NSMakeRange(1, typeStr.length - 2)];
+            ffi_type *type = malloc(sizeof(ffi_type));
+            type->alignment = 0;
+            type->size = 0;
+            type->type = FFI_TYPE_STRUCT;
+            NSDictionary *structDefine = [JPExtension registeredStruct][structName];
+            NSUInteger subTypeCount = [structDefine[@"keys"] count];
+            NSString *subTypes = structDefine[@"types"];
+            ffi_type **sub_types = malloc(sizeof(ffi_type *) * (subTypeCount + 1));
```

Comment:
Yes, you are right. It doesn't support complex struct types. I can complete it later.